File: mycollab-config/src/main/java/com/mycollab/configuration/IDeploymentMode.java
package com.mycollab.configuration;

/**
 * @author MyCollab Ltd
 * @since 5.2.2
 */
public interface IDeploymentMode {
    boolean isDemandEdition();

    boolean isCommunityEdition();

    boolean isPremiumEdition();

    String getSiteUrl(String subDomain);
}

File: mycollab-config/src/main/java/com/mycollab/configuration/IDeploymentMode.kt
package com.mycollab.configuration

/**
 * @author MyCollab Ltd
 * @since 5.2.2
 */
interface IDeploymentMode {
    val isDemandEdition: Boolean

    val isCommunityEdition: Boolean

    val isPremiumEdition: Boolean

    fun getSiteUrl(subDomain: String): String
}
